src/main/java/com/zendesk/maxwell/row/MaxwellJson.java

```diff
@@ -0,0 +1,96 @@
+package com.zendesk.maxwell.row;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.zendesk.maxwell.scripting.Scripting;
+import jdk.nashorn.api.scripting.ScriptObjectMirror;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.script.ScriptException;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.List;
+
+public class MaxwellJson {
+	private static final Logger LOGGER = LoggerFactory.getLogger(MaxwellJson.class);
+	private static final JsonFactory jsonFactory = new JsonFactory();
+
+	private final ByteArrayOutputStream buffer;
+	private final JsonGenerator jsonGenerator;
+	private final PlaintextJsonGenerator plaintextGenerator;
+	private final EncryptingJsonGenerator encryptingGenerator;
+
+	private static final ThreadLocal<MaxwellJson> instance =
+			new ThreadLocal<MaxwellJson>(){
+				@Override
+				protected MaxwellJson initialValue() {
+					try {
+						return new MaxwellJson();
+					} catch (IOException e) {
+						LOGGER.error("error initializing MaxwellJson", e);
+						return null;
+					}
+				}
+			};
```

Comment:
check out `ThreadLocal.withInitial(? -> ())` for a cleaner lambda implementation.

Reply:
Fancy! 😉